props/saber_BC_buttons.h

```diff
@@ -386,31 +1136,297 @@ EFFECT(volmax);     // for maximum volume reached
+    mode::getSL<SPEC>()->SayVolumeDown();
+  }
+
+  void QuickMaxVolume() {
+    dynamic_mixer.set_volume(max_volume_);
+    PVLOG_NORMAL << "** Maximum Volume\n";
+    mode::getSL<SPEC>()->SayMaximumVolume();
+  }
+
+  void QuickMinVolume() {
+    dynamic_mixer.set_volume(min_volume_);
+    PVLOG_NORMAL << "** Minimum Volume\n";
+    mode::getSL<SPEC>()->SayMinimumVolume();
+  }
+
+  void update() override {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    // Overridden to substitute the tick sound
+    this->say_time_ = Cyclint<uint32_t>(millis()) + 1;
+    if (!this->say_time_) this->say_time_ += 1;
+  }
+
+  void select() override {
+    PVLOG_NORMAL << "** Saved - Exit Volume Menu\n";
+    mode::getSL<SPEC>()->SaySave();
+    SPEC::SteppedMode::select();
+  }
+
+  void exit() override {
+    PVLOG_NORMAL << "** Cancelled - Exit Volume Menu\n";
+    percentage_ = initial_percentage_;
+    dynamic_mixer.set_volume(initial_volume_);
+    mode::getSL<SPEC>()->SayCancel();
+    SPEC::SteppedMode::exit();
+  }
+
+  bool mode_Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, 0)) {
+      // Custom button controls for BCVolumeMode
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_HELD_MEDIUM, 0):
+        QuickMaxVolume();
+        return true;
+
+      case EVENTID(BUTTON_POWER, EVENT_SECOND_HELD_MEDIUM, 0):
+      case EVENTID(BUTTON_AUX, EVENT_FIRST_HELD_MEDIUM, 0):
+        QuickMinVolume();
+        return true;
     }
+    // Use the select and exit controls from SelectCancelMode
+    return SPEC::SelectCancelMode::mode_Event2(button, event, modifiers);
+  }
+};
+
+#ifdef DYNAMIC_BLADE_LENGTH
+int selected_blade_;
+
+#ifdef BC_DUAL_BLADES
+template<class SPEC>
+struct BCSelectBladeMode : public SPEC::MenuBase {
+  int current_blade() { return this->pos_ + 1; }
+  ShowColorSingleBladeTemplate<Pulsing<ShowColorStyle,Black,800>,Pulsing<ShowColorStyle,Black,800>> highlighted_blade_;
+
+  uint16_t size() override {
+    return NUM_BLADES;
+  }
+  void mode_activate(bool onreturn) override {
+    mode::getSL<SPEC>()->SayEditBladeLength();
+    mode::getSL<SPEC>()->SaySelectBlade();
+    ShowColorStyle::SetColor(Color16(65535, 65535, 65535));
+    SPEC::SteppedMode::mode_activate(onreturn);
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+    say();
+  }
+
+  void mode_deactivate() {
+    highlighted_blade_.Stop(current_blade());
+  }
+
+  void next() override {
+    highlighted_blade_.Stop(current_blade());
+    SPEC::MenuBase::next();
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+    say();
+  }
+
+  void prev() override {
+    highlighted_blade_.Stop(current_blade());
+    SPEC::MenuBase::prev();
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+    say();
```

Comment:
I don't think this call to say() is needed.
Since you're inheriting from MenuBase, update() will be called after this function, which will call fadeout(), which will essentially cancel anything this call is doing. say() will be called after ccchange.wav  / beep is played.

Reply:
Indeed. Confirmed works fine without this. Removed.
I've also moved SayEditBladeLength() to the button event call instead, and removed "Blade" before the number.
It was too much, especially when exiting to have to hear each time "Edit blade length, select blade, blade 1".